clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -52,6 +65,127 @@ abstract class OpenstackClientProvider {
   }
 
   /**
+   * Gets a load balancer pool for a given region and load balancer UUID.
+   * @param region
+   * @param loadBalancerId
+   * @return
+   */
+  LbPool getLoadBalancerPool(final Region region, final String loadBalancerId) {
+    client.useRegion(region.name).networking().loadbalancers().lbPool().get(loadBalancerId)
+  }
+
+  /**
+   * Validates the subnet is valid in region.
+   * @param region
+   * @param subnetId
+   * @return boolean
+   */
+  boolean validateSubnetId (final Region region, final String subnetId) {
+    client.useRegion(region.name).networking().subnet().get(subnetId) == null
+  }
+
+  /**
+   * Creates a load balancer pool in provided region.
+   * @param region
+   * @param loadBalancerPool
+   * @return LbPool
+   */
+  LbPool createLoadBalancerPool(final Region region, final LoadBalancerPool loadBalancerPool) {
+    Protocol poolProtocol = Protocol.forValue(loadBalancerPool.protocol.name())
+    LbMethod poolMethod = LbMethod.forValue(loadBalancerPool.method.name())
+
+    getRegionClient(region).networking().loadbalancers().lbPool().create(
+      Builders.lbPool()
+        .name(loadBalancerPool.derivedName)
+        .protocol(poolProtocol)
+        .lbMethod(poolMethod)
+        .subnetId(loadBalancerPool.subnetId)
+        .description(loadBalancerPool.description)
+        .adminStateUp(Boolean.TRUE)
+        .build()
+    )
+  }
+
+  /**
+   * Creates a VIP for given region and pool.
+   * @param region
+   * @param virtualIP
+   * @return
+   */
+  Vip createVip(final Region region, final VirtualIP virtualIP) {
+    Protocol vipProtocol = Protocol.forValue(virtualIP.protocol.name())
+
+    getRegionClient(region).networking().loadbalancers().vip().create(Builders.vip()
+      .name(virtualIP.derivedName)
+      .subnetId(virtualIP.subnetId)
+      .poolId(virtualIP.poolId)
+      .protocol(vipProtocol)
+      .protocolPort(virtualIP.port)
+      .adminStateUp(Boolean.TRUE)
+      .build())
+  }
+
+  /**
+   * Creates a health check for given pool in specified region.
+   * @param region
+   * @param lbPoolId
+   * @param monitor
+   * @return
+   */
+  HealthMonitor createHealthCheckForPool(final Region region, final String lbPoolId, final PoolHealthMonitor monitor) {
+      HealthMonitor result = getRegionClient(region).networking().loadbalancers().healthMonitor().create(
+        Builders.healthMonitor().type(HealthMonitorType.forValue(monitor.type?.name()))
```

Comment:
The validator should ensure that the health check type is specified as it is a required field if a health check is present ... It isn't really required, but thought it would be better to have an IllegalArgumentException than an NPE.